app/javascript/components/lease_ups/LeaseUpApplicationsTable.js

```diff
@@ -107,14 +113,18 @@ const LeaseUpApplicationsTable = ({ listingId, dataSet, onLeaseUpStatusChange, o
       manual
       className='rt-table-status'
       data={dataSet}
+      page={currentPage}
       pages={pages}
       columns={columns}
       getTdProps={getTdProps}
       getTrProps={getTrProps}
       defaultPageSize={rowsPerPage}
       sortable={false}
       loading={loading}
-      onFetchData={onFetchData}
+      onFetchData={(state, instance) => {
+        setCurrentPage(state.page)
```

Comment:
can you explain a bit what's happening here? ReactTable has its own page in its state, right, so here we're saying, to update the react table's state to have the current page?

I found this answer for our issue, is that where you got the idea? https://spectrum.chat/react-table/general/pagination-reset-on-data-changed~2ef35446-d56a-4098-ac11-3e344f3af085

Reply:
Similar solution yes, I didn't actually see this prior, I read the SSR docs for the `manual` flag that we have set. We keep track of the page whenever data changes in the `onFetchData` call so we can set it by default with `useState` and then track it in state directly when the onFetch function fires